### PR TITLE
Extract gauge coloring and fix tests

### PR DESCRIPTION
**Pull request by audkar**
This fixes tests. `AlertsMvpController` manages `AlertsMvpView`. Gauge view is separate view. Clean solution would be to create separate controler for gauge view. Maybe I will refactor that later

### CODE REVIEW

File: app/src/main/java/net/kwatts/powtools/MainActivity.java [hunk 1]
```diff
@@ -367,12 +368,33 @@ public void onPropertyChanged(Observable observable, int i) {
...
     }
 
+    private void updateGaugeOnSpeedChange(ProgressiveGauge gauge, @NonNull String speedString) {
+        net.kwatts.powtools.util.SharedPreferences sharedPreferences = App.INSTANCE.getSharedPreferences();
+        float speed;
+        try {
+            speed = Float.parseFloat(speedString);
+        } catch (NumberFormatException e) {
+            Timber.e(e);
+            return;
```
Comment (ebabel) [inline app/src/main/java/net/kwatts/powtools/MainActivity.java]: I prefer to re-throw as a runtime error

File: app/src/main/java/net/kwatts/powtools/MainActivity.java [hunk 2]
```diff
@@ -367,12 +368,33 @@ public void onPropertyChanged(Observable observable, int i) {
...
+        }
+        float speedAlert = sharedPreferences.getSpeedAlert();
+        boolean isEnabled = sharedPreferences.getSpeedAlertEnabled();
+
+        if (isEnabled && speed >= speedAlert) {
```
Comment (ebabel) [inline app/src/main/java/net/kwatts/powtools/MainActivity.java]: I like pulling the gauge out, but unfortunate duplication of logic. Maybe this if check can live in a shared util?

File: app/src/main/java/net/kwatts/powtools/MainActivity.java (hunk 1 above)
Comment (audkar) [reply app/src/main/java/net/kwatts/powtools/MainActivity.java]: But then there is no point to catching this exception at all. Maybe then I should remove whole try/catch?

**Comment by audkar** [comment]
Is this what you meant?

**Pull request merged**